core/src/main/java/com/yahoo/oak/EntryArrayHeap.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+import sun.misc.Unsafe;
+
+import java.util.Arrays;
+
+public class EntryArrayHeap implements EntryArray {
+
+    // The array is initialized to 0 - this is important!
+    private final long[] entries;
+
+    // Number of primitive fields in each entry
+    private final int fieldCount;
+
+    // number of entries to be maximally held
+    private final int entryCount;
+
+    /**
+     * @param entryCount how many entries should this instance keep at maximum
+     * @param fieldCount the number of fields (64bit) in each entry
+     */
+    EntryArrayHeap(int entryCount, int fieldCount) {
+        this.fieldCount = fieldCount;
+        this.entryCount = entryCount;
+        this.entries = new long[entryCount * fieldCount];
+    }
+
+    /**
+     * Converts external entry-index and field-index to the field's direct memory address.
+     * <p>
+     * @param entryIndex the index of the entry
+     * @param fieldIndex the field (long) index inside the entry
+     * @return the field's offset in the array
+     */
+    private int entryOffset(int entryIndex, int fieldIndex) {
+        return entryIndex * fieldCount + fieldIndex;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int entryCount() {
+        return entryCount;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int fieldCount() {
+        return entryCount;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void clear() {
```

Comment:
Is the clear() usage is exercised as part of the tests?

Reply:
I don't remember where it is used. But this method was part of EntryArray before this PR.